examples/ui-demo/src/hooks/useSendUOsErc20Sponsorship.ts

```diff
@@ -0,0 +1,103 @@
+import { useMutation } from "@tanstack/react-query";
+import { type Chain, type Hex, type Address } from "viem";
+import { useToast } from "@/hooks/useToast";
+import { type AlchemyTransport } from "@account-kit/infra";
+import { useModularAccountV2Client } from "./useModularAccountV2Client";
+import { USDC_GAS_APPROVAL_AMOUNT } from "@/components/modals/Erc20/utils";
+import { DEMO_USDC_ADDRESS_6_DECIMALS } from "../utils/constants";
+
+const ERC20_SPONSORSHIP_POLICY_ID =
+  process.env.NEXT_PUBLIC_ERC20_SPONSORSHIP_POLICY_ID;
+
+export type UserOperationCall = {
+  target: Address;
+  data: Hex;
+  value?: bigint;
+};
+
+export interface UseSendUOsErc20SponsorshipParams {
+  toastText: string;
+  clientOptions: {
+    mode: "default" | "7702";
+    chain: Chain;
+    transport: AlchemyTransport;
+  };
+}
+
+export interface UseSendUOsErc20SponsorshipReturn {
+  isLoadingClient: boolean;
+  isSending: boolean;
+  txHash?: Hex;
+  error?: Error | null;
+  sendUOs: (userOperations: UserOperationCall[]) => void;
+  sendUOsAsync: (
+    userOperations: UserOperationCall[],
+  ) => Promise<Hex | undefined>;
+  reset: () => void;
+}
+
+export const useSendUOsErc20Sponsorship = (
+  params: UseSendUOsErc20SponsorshipParams,
+): UseSendUOsErc20SponsorshipReturn => {
+  const { clientOptions } = params;
+  const { setToast } = useToast();
+
+  const { client, isLoadingClient } = useModularAccountV2Client({
+    ...clientOptions,
+    policyId: ERC20_SPONSORSHIP_POLICY_ID,
+    policyToken: {
+      address: DEMO_USDC_ADDRESS_6_DECIMALS,
+      maxTokenAmount: USDC_GAS_APPROVAL_AMOUNT,
+    },
+  });
+
+  const {
+    mutate: sendUOs,
+    mutateAsync: sendUOsAsync,
+    data: txHash,
+    isPending: isSending,
+    error,
+    reset,
+  } = useMutation<Hex | undefined, Error, UserOperationCall[]>({
+    mutationFn: async (userOperations: UserOperationCall[]) => {
+      if (!client) {
+        throw new Error("Smart account client not ready");
+      }
+      if (!client.account) {
+        throw new Error("Smart account not connected or address not available");
+      }
+
+      const userOpHash = await client.sendUserOperation({
+        uo: userOperations,
+      });
+
+      return client.waitForUserOperationTransaction(userOpHash);
```

Comment:
not 100% sure how tanstack query catches errors, but errors from this return _might_ slip through if it's not awaited?
```suggestion
      return await client.waitForUserOperationTransaction(userOpHash);
```